src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2320,6 +2323,8 @@ public void EndConnect(IAsyncResult asyncResult)
             Exception? ex = castedAsyncResult.Result as Exception;
             if (ex != null || (SocketError)castedAsyncResult.ErrorCode != SocketError.Success)
             {
+                if (SocketsTelemetry.Log.IsEnabled()) SocketsTelemetry.Log.ConnectFailed((SocketError)castedAsyncResult.ErrorCode, ex);
```

Comment:
We want to measure timings as close to the wire as possible. It was requested on the issue #827

>The intent is to get these timings as low down in the stack and close as possible to the point that actual bytes are sent and received over the network so that accurate pictures can be developed